packages/mobile/src/transactions/send.ts

```diff
@@ -41,27 +42,48 @@ const getLogger = (tag: string, txId: string) => {
         break
       case SendTransactionLogEventType.EstimatedGas:
         Logger.debug(tag, `Transaction with id ${txId} estimated gas: ${event.gas}`)
-        CeloAnalytics.track(CustomEventNames.transaction_send_gas_estimated, { txId })
+        CeloAnalytics.track(CustomEventNames.transaction_send_gas_estimated, {
+          txId,
+          duration: new Date().getTime() - startTime,
+        })
         break
       case SendTransactionLogEventType.ReceiptReceived:
         Logger.debug(
           tag,
           `Transaction id ${txId} received receipt: ${JSON.stringify(event.receipt)}`
         )
-        CeloAnalytics.track(CustomEventNames.transaction_send_gas_receipt, { txId })
+        CeloAnalytics.track(CustomEventNames.transaction_send_gas_receipt, {
+          txId,
+          duration: new Date().getTime() - startTime,
+        })
         break
       case SendTransactionLogEventType.TransactionHashReceived:
         Logger.debug(tag, `Transaction id ${txId} hash received: ${event.hash}`)
+        CeloAnalytics.track(CustomEventNames.transaction_send_gas_hash_received, {
+          txId,
+          duration: new Date().getTime() - startTime,
+        })
         break
       case SendTransactionLogEventType.Started:
+        startTime = new Date().getTime()
         Logger.debug(tag, `Sending transaction with id ${txId}`)
         CeloAnalytics.track(CustomEventNames.transaction_send_start, { txId })
         break
       case SendTransactionLogEventType.Failed:
         Logger.error(tag, `Transaction failed: ${txId}`, event.error)
+        CeloAnalytics.track(CustomEventNames.transaction_error, {
+          txId,
+          error: event.error.message,
+          duration: new Date().getTime() - startTime,
+        })
         break
       case SendTransactionLogEventType.Exception:
         Logger.error(tag, `Transaction Exception caught ${txId}: `, event.error)
+        CeloAnalytics.track(CustomEventNames.transaction_exception, {
+          txId,
+          error: event.error.message,
```

Comment:
Repeating the feedback I left on Anna's PR. Are we using analytics for errors now? That's what Sentry was meant for and seems better suited to it.

Reply:
We're still using Sentry for errors, but we are also including them in analytics when we want to understand the error in context of what was going on in the app when the error was thrown

Discussion here- https://github.com/celo-org/celo-monorepo/pull/4144#discussion_r443681952